resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnAllocatorSuite.scala

```diff
@@ -523,9 +523,9 @@ class YarnAllocatorSuite extends SparkFunSuite with Matchers with BeforeAndAfter
     handler.getNumUnexpectedContainerRelease should be (2)
   }
 
-  test("blacklisted nodes reflected in amClient requests") {
-    // Internally we track the set of blacklisted nodes, but yarn wants us to send *changes*
-    // to the blacklist.  This makes sure we are sending the right updates.
+  test("excludeOnFailure nodes reflected in amClient requests") {
+    // Internally we track the set of excluded nodes, but yarn wants us to send *changes*
+    // to it.  This makes sure we are sending the right updates.
```

Comment:
I don't think the changes in `YarnAllocatorSuite` are correct -- I think this is testing the interaction with YARN's blacklisting feature (e.g. YARN-4576), rather than Spark's feature. I think it's the same for `YarnSchedulerBackendSuite` ?

Reply:
In this case actually this is what I wanted.  We internally track what nodes are excluded. This is from the excludeOnFailure list from the scheduler, from the spark.yarn.exclude.nodes config and the allocator logic for excluding nodes that have failures..  Yes we use the yarn blacklist list api to actually specify it to yarn but I don't see any reason internally we have to keep our code using the term blacklist except where we call into their api.  I could add a comment about that in there to clarify in the Allocator itself until they can update their api.
--

Reply:
Note the test case name actually should just be excluded nodes here so i'll update that.

Reply:
Makes sense -- I will take another look with this in mind. Thanks for explaining!